pkg/authorization/api/v1beta1/types.go

```diff
@@ -82,6 +82,38 @@ type PolicyBinding struct {
 	RoleBindings []NamedRoleBinding `json:"roleBindings"`
 }
 
+// ResourceAccessReviewSpec contains information about the action being requested.
+type ResourceAccessReviewSpec struct {
+	// Verb is one of: get, list, watch, create, update, delete
+	Verb string `json:"verb"`
+	// Resource is one of the existing resource types
+	Resource string `json:"resource"`
+	// Content is the actual content of the request for create and update
+	Content kruntime.RawExtension `json:"content,omitempty"`
+	// ResourceName is the name of the resource being requested for a "get" or deleted for a "delete"
+	ResourceName string `json:"resourceName,omitempty"`
```

Comment:
> You should include ResourceNamespace here.

Allowing a global resourceAccessReview that can then inspect who has access to resources in a different namespace makes authorization hard.  By forcing you to specify a namespace for the ResourceAccessReview and using that namespace for the request, it makes authorization rules easy to create, easy to understand, and mostly correct by default.